image_transport/test/test_message_passing_lifecycle.cpp

```diff
@@ -0,0 +1,183 @@
+  }
+
+  rclcpp_lifecycle::LifecycleNode::SharedPtr node_;
+};
+
+
+void imageCallback(const sensor_msgs::msg::Image::ConstSharedPtr & msg)
+{
+  (void) msg;
+  total_images_received++;
+}
+
+TEST_F(MessagePassingTestingLifecycle, one_message_passing)
+{
+  const size_t max_retries = 3;
+  const size_t max_loops = 200;
+  const std::chrono::milliseconds sleep_per_loop = std::chrono::milliseconds(10);
+
+  rclcpp::executors::SingleThreadedExecutor executor;
+
+  auto pub = image_transport::create_publisher(node_, "camera/image");
+  auto sub =
+    image_transport::create_subscription(node_, "camera/image", imageCallback, "raw");
+
+  test_rclcpp::wait_for_subscriber(node_->get_node_graph_interface(), sub.getTopic());
+
+  ASSERT_EQ(0, total_images_received);
+  ASSERT_EQ(1u, pub.getNumSubscribers());
+  ASSERT_EQ(1u, sub.getNumPublishers());
+
+  executor.spin_node_some(node_->get_node_base_interface());
+  ASSERT_EQ(0, total_images_received);
+
+  size_t retry = 0;
+  while (retry < max_retries && total_images_received == 0) {
+    // generate random image and publish it
+    pub.publish(generate_random_image());
+
+    executor.spin_node_some(node_->get_node_base_interface());
+    size_t loop = 0;
+    while ((total_images_received != 1) && (loop++ < max_loops)) {
+      std::this_thread::sleep_for(sleep_per_loop);
+      executor.spin_node_some(node_->get_node_base_interface());
+    }
+  }
+
+  ASSERT_EQ(1, total_images_received);
+}
+
+TEST_F(MessagePassingTestingLifecycle, one_camera_message_passing)
+{
+  const size_t max_retries = 3;
+  const size_t max_loops = 200;
+  const std::chrono::milliseconds sleep_per_loop = std::chrono::milliseconds(10);
+
+  rclcpp::executors::SingleThreadedExecutor executor;
+
+  auto pub = image_transport::create_camera_publisher(node_, "camera/image");
+  auto sub = image_transport::create_camera_subscription(
+    node_, "camera/image",
+    [](const sensor_msgs::msg::Image::ConstSharedPtr & image,
+    const sensor_msgs::msg::CameraInfo::ConstSharedPtr & info) {
+      (void) image;
+      (void) info;
+      total_images_received++;
+    },
+    "raw"
+  );
+
+  test_rclcpp::wait_for_subscriber(node_->get_node_graph_interface(), sub.getTopic());
+
+  ASSERT_EQ(0, total_images_received);
+  executor.spin_node_some(node_->get_node_base_interface());
+  ASSERT_EQ(0, total_images_received);
+
+  size_t retry = 0;
+  while (retry < max_retries && total_images_received == 0) {
+    // generate random image and publish it
+    pub.publish(*generate_random_image().get(), sensor_msgs::msg::CameraInfo());
+
+    executor.spin_node_some(node_->get_node_base_interface());
+    size_t loop = 0;
+    while ((total_images_received != 1) && (loop++ < max_loops)) {
+      std::this_thread::sleep_for(sleep_per_loop);
+      executor.spin_node_some(node_->get_node_base_interface());
+    }
+  }
+
+  ASSERT_EQ(1, total_images_received);
+}
+
+/*
+TEST_F(MessagePassingTestingLifecycle, stress_message_passing)
```

Comment:
remove ?

Reply:
I was simply referencing from test_message_passing.cpp, shall this be removed entirely?